pandas/core/apply.py

```diff
@@ -330,9 +331,18 @@ def agg_list_like(self) -> DataFrame | Series:
             # degenerate case
             if selected_obj.ndim == 1:
 
+                args_remain, kwargs_remain = self.args, self.kwargs
                 for a in arg:
                     colg = obj._gotitem(selected_obj.name, ndim=1, subset=selected_obj)
-                    new_res = colg.aggregate(a)
+                    if not isinstance(a, (np.ufunc, str)) and isinstance(
+                        a, FunctionType
+                    ):
+                        args_pass, kwargs_pass, args_remain, kwargs_remain = _map_args(
+                            a, args_remain, kwargs_remain
+                        )
+                        new_res = colg.aggregate(a, *args_pass, **kwargs_pass)
+                    else:
+                        new_res = colg.aggregate(a)
```

Comment:
I think we should always pass args, kwargs to the function, regardless of type and regardless of whether or not the function accepts them. If the user is passing things the function doesn't accept, then we should raise. Does this break something?

Reply:
When we pass `df.agg([foo1, foo2], 0, 5, b=6, c=7, d=8)`, `5, b=6` for `foo1` and `c=7, d=8` for `foo2`, the `self.args = (5,)` and `self.kwags={'b': 6, 'c': 7, 'd': 8}`. 

I think the key is how do we split args(`self.arg, self.kwargs`) for the list-like functions(`arg`), and pass them to the corresponding function(`a`).

The func `_map_args` is aim to spilt `arg_pass = (5,) kwargs_pass = {'b': 6}` for `foo1`, and remainder could pass to `foo2`.  

This is the botched method I found. But it is true that it will only ignore incorrect args instead raise error.

Reply:
> I think the key is how do we split args(`self.arg, self.kwargs`) for the list-like functions(`arg`), and pass them to the corresponding function(`a`).

I don't believe we should split args between functions. When a list is passed, I believe the args and kwargs should be passed to all functions in that list. E.g. `df.agg([foo1, foo2], 0, 5, b=6, c=7, d=8)` should pass `args=(5,)` and `kwargs={'b': 6, 'c': 7, 'd': 8}` to both `foo1` and `foo2`.

Reply:
I'm sorry I still don't fully understand what you're saying. 

When we pass `kwargs={'b': 6, 'c': 7, 'd': 8}` to `foo1` which `def foo1(a, b)`, how could we deal with the `TypeError` caused by unexpected keyword argument?  Raise? It doesn't seem quite right, as they are for the next `foo2`.

Reply:
> how could we deal with the `TypeError` caused by unexpected keyword argument? Raise? It doesn't seem quite right, as they are for the next `foo2`.

Yes - I think it should raise. If foo takes kwargs, then we can't tell if they whether or not they are for `foo`. Or perhaps the user has a typo in the kwargs that were provided. We'd have to guess.

I think `.agg([foo1, foo2], *args, **kwargs)` should take the results of `.agg(foo1, *args, **kwargs)` and `.agg(foo2, *args, **kwargs)` and combine the results. If the user provides functions which don't take the arguments provided, then it should raise. This way there is no guessing and no odd edge cases.

Reply:
When user call `df.agg([foo1, foo2], 0, 5, b=6, c=7, d=8)` for `def foo1(a, b), def foo2(c, d)`, raise an error cased by unexpected argument is what we want? 

In other words, when user call `.agg([foo1, foo2], *args, **kwargs)`, only if `args, kwargs` can be passed and uesd by both `foo1, foo2` at the same time, we let it work, otherwise an error should be raised.

The example below is what we want: 
```python
def foo1(x, a=1, c=0):
    print(f"foo1: {a}, {c}")
    return x

def foo2(x, b=2, c=0):
    print(f"foo2: {b}, {c}")
    return x

df = pd.DataFrame({'x': [1]})
df.agg([foo1, foo2], 0, 3, c=4)
# foo1: 3, 4
# foo2: 3, 4
```

Do I understand this properly?

Reply:
Yes - that is correct. Note that in order to accomplish this, we can just always pass *args and **kwargs to the function; no introspection of the functions and their arguments is necessary.